### PR TITLE
feat: add Hooks Extension Framework tooling for filters #2

### PR DESCRIPTION
**Pull request by mariajgrimaldi**
**Description:** 

This PR adds the tooling needed by the Hooks Extension Framework to execute filters. This project is described in more depth in the [OEP-50](https://github.com/edx/open-edx-proposals/pull/184).  

The tooling consists of a pipeline ([Pipeline tooling ADR](https://github.com/eduNEXT/openedx-filters/blob/ef4e5e093fb753539cd16729bc8f924ba964e7a5/docs/decisions/0003-hooks-filter-tooling-pipeline.rst)) that executes a list of functions pre-configured that are associated with the filter via Django settings ([Filters configuration ADR](https://github.com/eduNEXT/openedx-filters/blob/ef4e5e093fb753539cd16729bc8f924ba964e7a5/docs/decisions/0002-hooks-filter-config-location.rst)).

**Dependencies:** 
The architectural design records where the configuration and pipeline design used by the tooling are defined in Pull Request #2. 

**Testing instructions:**

1. Run `make lms-shell`
2. Inside the shell run: 
    `pip install -e git+https://github.com/eduNEXT/openedx-filters.git@MJG/filter_tooling#egg=filter_tooling`
3. Run `pip install -e https://github.com/eduNEXT/openedx-basic-hooks.git`, you can also use your own test library. Here are defined the functions that will be executed by a trigger. 
4. [Here's](https://github.com/eduNEXT/edx-platform/pull/212/files) a quick and dirty way to create and use filters on edx-platform. In the example we are using pre_register_filter that's soon to be added (PR pending).
5. (Optional) If you're using our demo and sample openedx-base-hooks, you can use our pre_register_filter. What you'll have to do is execute the logic that the filter extended, for example: 
- Go to register
- Enter data for registration and in the field `Year of birth` add 2020 (or any year after 2000).
- Then, an error will be displayed caused by the HookFilterException raised by our filter function defined in [this  location](https://github.com/eduNEXT/edx-platform/blob/06e45eb856574d3757e44ca8513afe78ef351079/openedx/core/djangoapps/user_authn/views/register.py#L531)

**Reviewers:**
- [ ] @morenol 
- [ ] @felipemontoya 

**Merge checklist:**
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Version bumped
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are squashed

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPI after tag-triggered build is 
      finished.
- [ ] Delete working branch (if not needed anymore)

### CODE REVIEW

File: manage.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env python
```
Comment (morenol) [inline manage.py:1]: Is this file needed?

File: openedx_filters/tests/test_pipeline.py [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+        function_with_exception = Mock(side_effect=HookFilterException)
+        function_without_exception = Mock(return_value=return_value)
+        get_functions_mock.return_value = [
+            function_with_exception,
+            function_without_exception,
+        ]
+
+        result = run_pipeline(self.trigger_name, **self.kwargs)
+
+        self.assertDictEqual(result, return_value)  # TODO: fix
```
Comment (morenol) [inline openedx_filters/tests/test_pipeline.py:108]: What do we need to fix?

**Review by morenol — COMMENTED** [review]
This looks great!  I added a couple of comments

File: openedx_filters/__init__.py [hunk 3]
```diff
@@ -1,5 +1,5 @@
 """
-Events of the openedx platform.
+File used to expose available filters.
```
[the remark this answers is not in the document]
Comment (mariajgrimaldi) [reply openedx_filters/__init__.py:2]: No, sorry. I was about to remove this

**Review by felipemontoya — APPROVED** [review]
I tested this with different filter definitions, different pipeline configurations and different logic for raising exceptions and it performed as we have so long discussed via all the channels.

Thanks @mariajgrimaldi, I think we are ready to merge

**Comment by felipemontoya** [comment]
Please squash the 10 commits before merging
